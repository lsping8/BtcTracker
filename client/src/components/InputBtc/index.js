import React, { Component } from 'react';

class InputBtc extends Component {
  constructor(){
    super()
    this.state = {
      calPrice: 0,
      value: ''
    }
  }

  handleOnChange = (event) => {
    this.setState({ value: event.currentTarget.value, calPrice: this.calculatePrice(event.currentTarget.value)})
  }

  calculatePrice = (value) => {
    const { currentPrice } = this.props
    return parseFloat((value * currentPrice).toFixed(5))
  }

  render() {
    const { calPrice,value } = this.state
    const { currency } = this.props
    return (
      <div className="container-grid">
        <div className="box-input" style={{padding: 10,lineHeight: '40px'}}>
          <label>
            <input className="input-style" type="number" placeholder="Input your BTC" value={value} onChange={this.handleOnChange}/>
            <span> BTC</span>
          </label>
          <div/>
        </div>
        <div style={{margin: 'auto'}}>
          =
        </div>
        <div className="box" style={{padding: 10,lineHeight: '40px'}}>
           {currency} {calPrice === 0 ? 0 : this.calculatePrice(value)}
        </div>
      </div>
    );
  }
}

export default InputBtc;
