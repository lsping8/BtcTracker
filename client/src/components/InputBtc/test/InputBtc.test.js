import React from 'react'
import ReactDOM from 'react-dom'
import { shallow,configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import InputBtc from '../index'

configure({ adapter: new Adapter() });

const props = { currency: 'USD' }
const inputBtc = shallow(<InputBtc {...props}/>)

describe('InputBtc', () => {
  it('renders without crashing', () => {
    expect(inputBtc).toMatchSnapshot()
  })

  it('initializes the `state`', () => {
    expect(inputBtc.state().calPrice).toEqual(0)
    expect(inputBtc.state().value).toEqual('')
  })

  it('initial calculate price', () => {
    expect(inputBtc.find('.box').text()).toEqual('USD 0')
  })
})
