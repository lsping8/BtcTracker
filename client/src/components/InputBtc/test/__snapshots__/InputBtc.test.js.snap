// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBtc renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputBtc
    currency="USD"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="box-input"
          style={
                    Object {
                              "lineHeight": "40px",
                              "padding": 10,
                            }
          }
>
          <label>
                    <input
                              className="input-style"
                              onChange={[Function]}
                              placeholder="Input your BTC"
                              type="number"
                              value=""
                    />
                    <span>
                               BTC
                    </span>
          </label>
          <div />
</div>,
        <div
          style={
                    Object {
                              "margin": "auto",
                            }
          }
>
          =
</div>,
        <div
          className="box"
          style={
                    Object {
                              "lineHeight": "40px",
                              "padding": 10,
                            }
          }
>
          USD
           
          0
</div>,
      ],
      "className": "container-grid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              <input
                            className="input-style"
                            onChange={[Function]}
                            placeholder="Input your BTC"
                            type="number"
                            value=""
              />
              <span>
                             BTC
              </span>
</label>,
            <div />,
          ],
          "className": "box-input",
          "style": Object {
            "lineHeight": "40px",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="input-style"
                  onChange={[Function]}
                  placeholder="Input your BTC"
                  type="number"
                  value=""
/>,
                <span>
                   BTC
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "input-style",
                  "onChange": [Function],
                  "placeholder": "Input your BTC",
                  "type": "number",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " BTC",
                },
                "ref": null,
                "rendered": " BTC",
                "type": "span",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "=",
          "style": Object {
            "margin": "auto",
          },
        },
        "ref": null,
        "rendered": "=",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "USD",
            " ",
            0,
          ],
          "className": "box",
          "style": Object {
            "lineHeight": "40px",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          "USD",
          " ",
          0,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="box-input"
            style={
                        Object {
                                    "lineHeight": "40px",
                                    "padding": 10,
                                  }
            }
>
            <label>
                        <input
                                    className="input-style"
                                    onChange={[Function]}
                                    placeholder="Input your BTC"
                                    type="number"
                                    value=""
                        />
                        <span>
                                     BTC
                        </span>
            </label>
            <div />
</div>,
          <div
            style={
                        Object {
                                    "margin": "auto",
                                  }
            }
>
            =
</div>,
          <div
            className="box"
            style={
                        Object {
                                    "lineHeight": "40px",
                                    "padding": 10,
                                  }
            }
>
            USD
             
            0
</div>,
        ],
        "className": "container-grid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                <input
                                className="input-style"
                                onChange={[Function]}
                                placeholder="Input your BTC"
                                type="number"
                                value=""
                />
                <span>
                                 BTC
                </span>
</label>,
              <div />,
            ],
            "className": "box-input",
            "style": Object {
              "lineHeight": "40px",
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="input-style"
                    onChange={[Function]}
                    placeholder="Input your BTC"
                    type="number"
                    value=""
/>,
                  <span>
                     BTC
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input-style",
                    "onChange": [Function],
                    "placeholder": "Input your BTC",
                    "type": "number",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " BTC",
                  },
                  "ref": null,
                  "rendered": " BTC",
                  "type": "span",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "=",
            "style": Object {
              "margin": "auto",
            },
          },
          "ref": null,
          "rendered": "=",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "USD",
              " ",
              0,
            ],
            "className": "box",
            "style": Object {
              "lineHeight": "40px",
              "padding": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            "USD",
            " ",
            0,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
